assets/js/util/copy-to-clipboard.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copy to Clipboard utility.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Copies the given string of text to the clipboard.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} text Text to copy.
+ */
+export default function copyToClipboard( text ) {
+	const el = document.createElement( 'textarea' );
+	el.innerHTML = text;
+	document.body.appendChild( el );
```

Comment:
Should this have some kind of CSS rule to be hidden? Theoretically it could become visible for a split second. May not be a big deal, but it could create a strange experience.

Reply:
@aaemnnosttv ^

Reply:
The short answer is I think this should be okay; it should happen too fast for it to ever be visible. FWIW, this is how it's done on Jurassic Ninja.

The long answer is that I looked into this a bit, and it's somewhat complicated. WP originally had a similar approach for use with the Site Health copy button, but replaced it with clipboardjs.com (see https://github.com/WordPress/WordPress/commit/65f7cdbce053c7ef0bd3356fc26432d379c1045d). This library boasts very good browser support, but isn't well-suited for copying arbitrary text, and also isn't well-suited for use in React due to added event listeners, and its `Clipboard` API instances that it creates. Not impossible but not ideal.

I think the need to hide the element is only relevant when it is persistent, like how WP had it implemented originally.

I've gone ahead and refactored the use here to use https://github.com/feross/clipboard-copy, which is a super-lightweight version of the same function, but it uses the browser's clipboard API if present first, which requires no temporary element, so I think this is the best solution for our needs.